src/components/SwapWidget/SwapWidget.tsx

```diff
@@ -165,6 +168,14 @@ const SwapWidget: FC = () => {
   const [isSwapping, setIsSwapping] = useState(false);
   const [isWrapping, setIsWrapping] = useState(false);
   const [isRequestingQuotes, setIsRequestingQuotes] = useState(false);
+  // if `bestTradeOption` is true, `isQueryingSelectedServer` will get toggled to false. This is so InfoSection.tsx renders the correct text & data
+  const [isQueryingSelectedServer, setIsQueryingSelectedServer] =
+    useState(false);
+
+  // Check if URL query string contains server URL
+  const location = useLocation();
+  const query = new URLSearchParams(location.search);
+  const queryUrl = query.get("serverURL");
```

Comment:
Make a custom `useSearchParams` hook for this so you end up with this:

```suggestion
const serverUrl = useSearchParams<{ serverUrl: string }>();
```

Reply:
I made a new file with the hook `useSearchParams`. Pushed it to this branch

Reply:
Now the dispatch in `useSearchParams` runs every render. Please install redux dev tools so you can catch this kind of mistakes.

Again, please make a `useSearchParams` that's reusable, so you can use it like this, with a custom generic:

`const serverUrl = useSearchParams<{ serverUrl: string }>();`

Reply:
What should the purpose of `useSearchParams` be? I'm trying to use it as a hook that return the query string from the URL. If the hook takes in a string as an argument, then it won't return a query string. It's purpose would then be to call dispatch and set the query string into state in Redux. If the hook calls dispatch, then it will be continuously called in the SwapWidget component. Not sure I'm understanding this